Moving Microworlds pages to scratch-www & adding arts

morantsur opened the pull request:
Moved https://scratch.mit.edu/pathways/hiphop and fashion, and created a new microworld for arts. 
The final design is not done yet, but most of the code is already in place.

Moved from #344

mewtaylor commented on src/components/tipsslider/tipsslider.json:
```diff
@@ -0,0 +1,50 @@
+[
```
It might be helpful to have the json in the format that microworlds use as a sample – i.e.  carousel items chunked into sets.

mewtaylor commented on src/components/tipsslider/tipsslider.jsx:
```diff
@@ -0,0 +1,64 @@
+require('slick-carousel/slick/slick-theme.scss');
+require('./tipsslider.scss');
+
+var TipsSlider = React.createClass({
```
small nitpick – it could be good to call it something like `NestedCarousel`, or in some way that describes a bit how it differs from the other Carousel component.

mewtaylor commented on src/components/tipsslider/tipsslider.jsx:
```diff
@@ -0,0 +1,64 @@
+                               src={item.thumbnailUrl} />)
+            }
+            stages.push(
+            <div className="testing" key={"stage_" + i}>
```
we can probably change this class name :)

mewtaylor commented on src/components/tipsslider/tipsslider.scss:
```diff
@@ -0,0 +1,75 @@
+    $button-offset: $icon-size + 5px;
+    $box-content-offset: 20px;
+
+    //padding: 12px $button-offset;
```
can probably remove this

mewtaylor commented on src/views/microworld_art/microworld_art.jsx:
```diff
@@ -0,0 +1,4 @@
+var render = require('../../lib/render.jsx');
+var Microworld = require('../../components/microworld/microworld.jsx');
+
+render(<Microworld microworldData={require("./microworld_art.json")} />, document.getElementById('view'));
```
I think the common convention for react is to require ahead of time to get the object, and then load it into the render method later on. @rschamp is that correct as far as you know?

mewtaylor commented on src/components/microworld/microworld.jsx:
```diff
@@ -0,0 +1,251 @@
+            videoOpen: {},
+            featuredGlobal: {},
+            featuredLocal: {},
+            microworld_data: this.props.microworldData
```
I think this could be moved to props instead of state if it's not going to be changed within the component itself

mewtaylor commented on src/components/microworld/microworld.jsx:
```diff
@@ -0,0 +1,251 @@
+            'top-banner'
+        );
+
+        var microworldData = this.state.microworld_data;
```
You could probably use `this.props.microworld_data.<...>` down below, which means you could get rid of this assigment

mewtaylor commented on src/components/microworld/microworld.scss:
```diff
@@ -0,0 +1,389 @@
+      border-left: 1px solid #E0E0E0;
+      background-color: white;
+      margin-bottom: 50px;
+      /* offset-x | offset-y | blur-radius | color */
```
could probably be deleted

mewtaylor commented:
Just a few comments, but this is awesome – I love how easy it makes it to prototype new pathways for people who are new to the code base! Besides the comments above, it might help to add some test cases to the PR description if there are any specific user behaviors that we want to check work correctly (i.e. videos showing/hiding, carousel item navigation, etc.)

Also, it looks like some tests are failing too, so it might make sense to run `make test` locally to see what might be failing.

thisandagain commented:
This is really wonderful @morantsur ! Excited to be able to prototype new Microworlds so easily.

/cc @ntlrsk @kaschm

morantsur commented:
Fixed all the comments, and removed some unneeded vars, ids and such.
I also changed the thumbnail component, to allow thumbnails without a link (used with the tips section).

I looked the the scss file for the tipsslider/nestedcarousel, and realized it was just a copy of the carousel scss, so I removed all the duplicate code - let me know if my solution makes sense. 

Where should I add the test cases? 

Thanks!

thisandagain commented:
@mewtaylor Could you follow-up on this with @morantsur ?

mewtaylor commented:
@thisandagain yes apologies for the delay!

mewtaylor commented:
@morantsur as discussed in person – if we can get the `nestedcarousel` to work with flex so that it's justified to the center, that would be great. Otherwise, looks great!

morantsur commented:
@mewtaylor 
I fixed the nested carousel issue, by just changing a few param and css of the slider, merged the code and added two changes I would like to get your OK for:
* Expanded that no-link option to include also the image itself (as it was an empty link before)
* I removed the routes to the microworlds, so for now they are detached. Once we decide it's good enough we can add them back. 

Thanks!

mewtaylor commented:
This looks awesome! :shipit:

rschamp commented:
@mewtaylor Talked with @thisandagain who thinks we should merge, but not set up the Fastly rules for this.  That will make it so this won't get stale, but also won't be released yet.
